apps/system/js/updatable.js

```diff
@@ -19,6 +21,49 @@
  * - cancelDownload() to cancel it
  */
 
+// Wrapper to read a settings value
+var setting = (function() {
+  var values = {};
+  return key => {
+
+    // Cache the promise straight away so multiple callers dont
+    // create multiple listeners
+    if (key in values) {
+      return values[key];
+    }
+
+    values[key] = new Promise(resolve => {
+      // We store the value of the setting and the observer is
+      // in charge of keeping it up to date
+      function cacheAndResolve() {
+        values[key] = (key in req.result) ? req.result[key] : undefined;
+        resolve(values[key]);
+      }
+
+      var lock = navigator.mozSettings.createLock();
+      var req = lock.get(key);
+
+      req.onerror = cacheAndResolve.bind(this);
+      req.onsuccess = cacheAndResolve.bind(this);
+
+      navigator.mozSettings.addObserver(key, e => {
+        values[key] = Promise.resolve(e.settingValue);
+      });
+    });
+
+    return values[key];
+  };
+})();
+
+// Should we automatically update?
+AppUpdatable.AUTO_UPDATE = 'addons.auto_update';
+
+// Notify the users after we have auto updated?
+AppUpdatable.NOTIFY_UPDATE = 'addons.update_notify';
```

Comment:
Yeh this is a seperate notification, there is a notification on the first auto update to happen, similiar to private tracking and that is stored in localstorage. There is a setting to let users be notified after every update and this keeps track of that